src/util/exec-in-repo.ts

```diff
@@ -8,11 +8,17 @@ interface IExecRepoResult {
   childProcess: ChildProcess;
 }
 
-export default (context: IMigrationContext, repo: IRepo, command: string): IExecRepoResult => {
+export default async (
```

Comment:
This was previously sync, requiring that the potentially long running `spawn` promise was returned to be able to be awaited at a later point. 

I kept that behavior the same, but need to get the environment variables first. So this method essentially now lets the implementation await some things, but not the spawn. Neat.

## My other approach
I initially tried to keep this method sync and put the async `getEnvrionmentVariables` call inside the promise by doing something like this:

```
const promise = adapterEnvironmentVarsPromise.then(
    (adapterEnvironmentVars: IEnvironmentVariables) => ({
        cwd: repoDir,
        env: {
          ...process.env,
          SHEPHERD_REPO_DIR: repoDir,
          SHEPHERD_DATA_DIR: dataDir,
          SHEPHERD_MIGRATION_DIR: migrationDir,
          SHEPHERD_BASE_BRANCH: baseBranch,
          ...adapterEnvironmentVars,
        },
        shell: true,
        capture: ['stdout', 'stderr'],
      })
  ).then((execOptions: import("child_process").SpawnOptions) => {
    return spawn(command, [], execOptions)
  }) as ChildProcessPromise;
```

However, the return of this function is 
```
return {
  promise,
  childProcess: promise.childProcess,
};
```
and the spawn promise won't have been created at the time this function returns, so `childProcess` doesn't exist on the `promise` variable.

Alas, I ended up to where I am in this PR.

*Note: there is only one callsite of this method, so there doesn't seem to be much risk of converting this to be async*

Reply:
The interface is a little awkward, but I think what you have is fine.